relay_messages_test.go

```diff
@@ -168,45 +186,34 @@ const (
 	resTotalCombinations
 )
 
-func (cr testCallRes) res() lazyCallRes {
-	f := NewFrame(100)
-	fh := FrameHeader{
-		size:        uint16(0xFF34),
-		messageType: messageTypeCallRes,
-		ID:          0xDEADBEEF,
+func (cr testCallRes) res(tb testing.TB) lazyCallRes {
+	params := testCallResParams{
+		payloadSize: 100,
 	}
-	f.Header = fh
-	fh.write(typed.NewWriteBuffer(f.headerBuffer))
-
-	payload := typed.NewWriteBuffer(f.Payload)
-
-	if cr&resIsContinued == 0 {
-		payload.WriteSingleByte(0) // flags
-	} else {
-		payload.WriteSingleByte(hasMoreFragmentsFlag) // flags
+	if cr&resIsContinued != 0 {
+		params.flags |= hasMoreFragmentsFlag
 	}
-
 	if cr&resIsOK == 0 {
-		payload.WriteSingleByte(1) // code not ok
-	} else {
-		payload.WriteSingleByte(0) // code ok
+		params.code = 1
 	}
-
-	headers := make(map[string]string)
 	if cr&resHasHeaders != 0 {
-		addRandomHeaders(headers)
+		kvList := []keyVal{
+			{"k1", "v1"},
+			{"k222222", ""},
+			{"k3", "thisisalonglongkey"},
+		}
```

Comment:
We do something different for call req transport headers (use a `map[string]string`, and then call `writeHeaders`), curious why we treat headers differently for call res.

Remember that these are the transport headers, and the headers returned from services will typically be in the arg2 for the response

Reply:
yea, i wanted to avoid the map so that the order of the entries is deterministic and easier for testing. but for the sake of consistency, I guess we should keep it the same.